### PR TITLE
Specialize client

### PR DESCRIPTION
**Pull request by gaschler**
More compile time checks.

### CODE REVIEW

File: async_grpc/client.h [hunk 1]
```diff
@@ -84,119 +77,221 @@ class Client {
...
-              ::grpc::internal::RpcMethod::NORMAL_RPC ||
-          rpc_method_.method_type() ==
-              ::grpc::internal::RpcMethod::CLIENT_STREAMING);
-    return response_;
+  ::grpc::Status StreamFinish() {
+    InstantiateClientWriterIfNeeded();
+    return client_writer_->Finish();
   }
 
+  const ResponseType &response() { return response_; }
```
Comment (pifon2a) [inline async_grpc/client.h]: super-nit: ` const ResponseType& response() `to be consistent with the rest of code base. Clang-format can change it back though, if it finds another ref styled like that first.

**Review by pifon2a — COMMENTED** [review]
very good.

**Review by pifon2a — APPROVED** [review]

File: async_grpc/client.h (hunk 1 above)
Comment (gaschler) [reply async_grpc/client.h]: Done

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Pull request merged**